lib/beaker-pe/install/pe_utils.rb

```diff
@@ -450,7 +450,15 @@ def do_install hosts, opts = {}
               else
                 prepare_host_installer_options(host)
                 register_feature_flags!(opts)
-                generate_installer_conf_file_for(host, hosts, opts)
+
+                if opts[:upgrade] && use_meep?(host['previous_pe_ver'])
```

Comment:
Oh, I guess just checking `opts[:upgrade]` worked because I was testing upgrades from 2017.1 and that version just ignored the next flag.

As for the upgrade cases, that matches up with my expectations. Perhaps I should make that a comment for future us.